packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -1367,6 +1367,17 @@ export class LocalIntervalCollection<TInterval extends ISerializableInterval> {
 			}
 
 			if (props) {
+				// This check is intended to prevent scenarios where a random interval is created and then
+				// inserted into a collection. The aim is to ensure that the collection is created first
+				// then the user can create/add intervals based on the collection
+				if (
```

Comment:
nit: I don't think our error message here needs to be particularly friendly. `assert`ing here and below might give better bundle size. not a big deal though.